src/test/run-make-fulldeps/resolve-rename/Makefile

```diff
@@ -0,0 +1,7 @@
+-include ../tools.mk
+
+all:
+	$(RUSTC) -C extra-filename=-hash foo.rs
+	$(RUSTC) bar.rs
```

Comment:
If you pass `--crate-name foo` here, does that make the test fail on today's compiler?

Reply:
It does, but because it's identical to one of its dependencies (same crate name and metadata), which is also a failure with this patch.

Reply:
Could a different metadata be passed to `bar.rs` here so that error path wouldn't get hit?

Reply:
Yeah, but then we fail because `baz.rs` can't tell which version of `foo` to use, which is still an issue with this patch because we don't yet have an extra-filename for `foo`, because we're searching for it as a direct dependency of `baz`. This seemed plausible:

```
$(RUSTC) -C extra-filename=-hash foo.rs
$(RUSTC) --crate-name foo -C metadata=baz -C extra-filename=-another-hash bar.rs
$(RUSTC) --extern foo=$(TMPDIR)/libfoo-another-hash.rlib baz.rs
$(RUSTC) qux.rs
```

where `qux` depends on `baz`, but even then `locator.rs` doesn't produce an error because it can differentiate between `foo` crates by the SVH it finds in the metadata for deps of `baz`, and doesn't consider the `foo` found in `foo.rs` when resolving dependencies for `baz.rs`.

After trying this and a few more things I'm sort of convinced we're not going to get further in this direction, because whenever `extra_filename` is present in the locate context, `hash` will be too, and in this case `locator.rs` doesn't produce the kind of errors we're trying to induce.

Should we consider landing this without the test?

Reply:
Ah I think you're right, that makes sense to me and means that it may be quite hard to add a test. In that case I think it's fine, do you want to remove this one as it's not testing what we thought it was testing?

Reply:
This test may still be interesting, because it's a case that fails with an earlier version of this patch (which didn't fall back to an imprecise match if the match considering `extra-filename` failed), but I'd be ok with leaving it out for now as well. Thanks!

Reply:
Oh ok! If it catches a previous mistake sounds good to leave in